### PR TITLE
Make some improvements for building on MinGW.

### PR DESCRIPTION
**Pull request by wbhart**
* set prefix on MinGW
* put dll in bin
* make import library
* fix install and uninstall targets for MinGW
 
Could someone take a look at this and see if I missed anything:

@fredrik-johansson @albinahlback @tthsq12 @isuruf

Please ignore PR on @fredrik-johansson 's repo. This is the correct one here.

### CODE REVIEW

**Comment by albinahlback** [comment]
Sorry, I don't think I can bring anything to the table for this one.

**Comment by wbhart** [comment]
Thanks @isuruf 

It seems to work now without making any additional copies of files about the place. Both `make install` and `make uninstall` now seem to work correctly on both Linux and MinGW64. So hopefully it is all correct now.

Let me know if you spot anything else wrong. If not I'll merge it tomorrow and hopefully issue the final RC for Flint-2.9.

**Comment by wbhart** [comment]
Let's see if the CI still passes.

File: configure [hunk 1]
```diff
@@ -486,13 +498,23 @@ if [ -z "$FLINT_LIB" ]; then
          EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -compatibility_version $FLINT_MAJOR.$FLINT_MINOR"
          EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -current_version $FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH"
          ;;
-      CYGWIN* | MINGW*)
+      CYGWIN*)
+         FLINT_LIBNAME="libflint.dll"
```
Comment (isuruf) [inline configure:502]: An unversioned DLL is unnecessary. You only need the unversioned import library (libflint.dll.a)

File: configure [hunk 2]
```diff
@@ -486,13 +498,23 @@ if [ -z "$FLINT_LIB" ]; then
...
+         FLINT_LIB="libflint-$FLINT_MAJOR.dll"
+         EXTRA_SHARED_FLAGS="-static-libgcc"
+         EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -shared"
+         EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -Wl,--export-all-symbols"
+         EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -Wl,-soname,libflint-$FLINT_MAJOR.dll.$FLINT_MINOR.$FLINT_PATCH"
```
Comment (isuruf) [inline configure:507]:
```suggestion
         EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -Wl,-soname,libflint-$FLINT_MAJOR.dll"
```

Comment (wbhart) [reply configure:507]: If the line numbers here are correct, this changes only Cygwin. Should I make the same change for MinGW as well?

File: configure (hunk 1 above)
Comment (wbhart) [reply configure:502]: Again, should I make the same change on MINGW too?

Comment (isuruf) [reply configure:502]: I don't think there's a need to have two different schemes for `MinGW` and `Cygwin`

File: configure (hunk 2 above)
Comment (isuruf) [reply configure:507]: On second thought, this whole line is unnecessary.

File: configure (hunk 1 above)
Comment (isuruf) [reply configure:502]: You need `-Wl,--out-implib,libflint.dll.a`. I don't think `-Wl,--subsystem,windows` is needed even for MinGW.

> I thought Cygwin didn't use native dlls at all?

They do. See for example https://cygwin.com/packages/x86_64/libgmp-devel/libgmp-devel-6.2.0-2
and https://cygwin.com/packages/x86_64/libgmp10/libgmp10-6.2.0-2

Comment (wbhart) [reply configure:502]: Is there a way to test a string againt CYGWIN* and MINGW* in the Makefile so I don't need to test for four different strings?

Comment (isuruf) [reply configure:502]: Other than setting a variable like `IS_WIN` in configure, I don't think so

Comment (wbhart) [reply configure:502]: Also, do you know where the import library is supposed to be installed on Cygwin? Does it just go in lib like the dll?

Comment (wbhart) [reply configure:502]: Oh that's bizarre. How did our Cygwin or MinGW CI ever pass before?

Comment (wbhart) [reply configure:502]: I don't suppose you know what prefix should be by default on Cygwin32 and Cygwin64?

**Comment by wbhart** [comment]
@isuruf I think it should do the same on Cygwin and MinGW now, if you want to have one final look.

I tested `make`, `make install` and `make uninstall` on Linux and MinGW64.

Currently prefix is not given a default on Cygwin as I don't know what that should be (it's '/mingw64' or '/mingw32' on MSYS2.

**Comment by wbhart** [comment]
Ok, according to [1], Cygwin tries to emulate a Linux path structure. So I'll leave the default prefix alone on Cygwin.


I'm going to merge this PR now and propagate to flint-2.9 and release the next rc.

[1] http://www.voxforge.org/home/docs/cygwin-cheat-sheet

**Pull request merged**